pkg/processor/trigger/kinesis/trigger.go

```diff
@@ -79,16 +83,13 @@ func newTrigger(parentLogger logger.Logger,
 func (k *kinesis) Start(checkpoint functionconfig.Checkpoint) error {
 	k.Logger.InfoWith("Starting",
 		"streamName", k.configuration.StreamName,
-		"shards", k.configuration.Shards)
+		"applicationName", k.configuration.ApplicationName)
 
-	for _, shardInstance := range k.shards {
-
-		// start reading from shard
-		go func(shardInstance *shard) {
-			if err := shardInstance.readFromShard(); err != nil {
-				k.Logger.ErrorWith("Failed to read from shard", "err", err)
-			}
-		}(shardInstance)
+	for _, workerInstance := range k.kclWorkers {
```

Comment:
1. workerInstance -> kclWorkerInstance (consistency)
2. There can be many kclWorkers and I imagine Start() is blocking. Perhaps start them using an errgroup? This means all will start in parallel

Reply:
It doesn't seem to be blocking. It runs the event loop in a go routine.